sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/variant/variantExpressions.scala

```diff
@@ -248,9 +253,10 @@ case object VariantGet {
       input: VariantVal,
       dataType: DataType,
       failOnError: Boolean,
-      zoneId: Option[String]): Any = {
+      zoneStr: Option[String],
```

Comment:
why would we need zone id string during execution?

Reply:
The zone id string is needed to construct the `Cast` object. It can also be obtained from `zoneId` (`zoneStr == Some(zoneId.getId)`), but that would require more runtime cost than passing both the string and the id.